src/mqtt/MQTT.cpp

```diff
@@ -161,8 +162,11 @@ void MQTT::onReceive(char *topic, byte *payload, size_t length)
                         p->channel = ch.index;
                     }
 
+                    // PKI messages get accepted even if we can't decrypt
+                    if (router && p->which_payload_variant == meshtastic_MeshPacket_encrypted_tag && p->channel == 0)
```

Comment:
Better to check for `(strcmp(e.channel_id, "PKI")`  to avoid not decrypting when you happen to have a channel hash of 0.

Also, I think here we want to check whether the `to` and `from` are in our NodeDB to avoid routing PKI messages for nodes that do not talk via your channels.

Reply:
Latest change to 2.5 branch fixes all of this, except checking for to/from. I'm still pondering that one.